src/main/java/org/cactoos/iterable/Matched.java

```diff
@@ -65,13 +68,14 @@ public Matched(final Iterable<X> fst, final Iterable<X> snd) {
      * @param snd The second part of duplex iterator.
      */
     public Matched(
-        final BiFunc<X, X, Boolean> fnc,
-        final Iterable<X> fst, final Iterable<X> snd
+        final BiFunc<? super X, ? super X, Boolean> fnc,
+        final Iterable<? extends X> fst,
+        final Iterable<? extends X> snd
```

Comment:
@andreoss let's replace the type of `snd` with `Iterable<? extends Y>` with Y being a type declared at the constructor level (or you can just add a todo). There is no need for this second parameter to be so constrained in types.

Reply:
@victornoel Done